src/bot/integrations/qiwi.ts

```diff
@@ -0,0 +1,113 @@
+import axios from 'axios';
+import { onChange, onStartup } from '../decorators/on.js';
+import { settings } from '../decorators';
+import { ui } from '../decorators.js';
+import { triggerInterfaceOnTip } from '../helpers/interface/triggers';
+import { error, tip } from '../helpers/log';
+import Integration from './_interface';
+
+class Qiwi extends Integration {
+  interval: any = null;
+
+  @settings()
+  @ui({ type: 'text-input', secret: true })
+  secretToken = '';
+
+  @onStartup()
+  @onChange('enabled')
+  onEnabledChange (key: string, val: boolean) {
+    if (val) {
+      this.start();
+    } else {
+      clearInterval(this.interval);
+    }
+  }
+
+  @onChange('secretToken')
+  onTokenChange (key: string, val: string) {
+    if (val) {
+      this.start();
+    } else {
+      clearInterval(this.interval);
+    }
+  }
+
+  async start () {
+    clearInterval(this.interval);
+    if (this.secretToken.trim() === '' || !this.enabled) {
+      return;
+    } else {
+      this.interval = setInterval(() => this.check(), 3000);
```

Comment:
Isn't 3000 too fast? I think minute or half minute should be OK

Reply:
No, it's ok. They haven't rate limits. i asked platform support about that.